### DISCUSSION TITLE
How do I store a Discord channel as a global variable?

### DISCUSSION THREAD

**Discussion by MasterOpel** [Rapptz/discord.py#8105] [General] [answered]
In my Discord bot, I need to store the Discord channel of specific messages as a global variable. My bot is [The Auto-Bumper Bot](https://github.com/Myth1calCode/Auto-Bumper-Discord-bot).

**Comment by rtk-rnjn** [comment, marked as answer]
I don't know who suggested you to use `Thread`. `discord.py` has its [own task system](https://discordpy.readthedocs.io/en/latest/ext/tasks/index.html) \*shrug*

You can define partial channel and store it in bot instance (it works same as channel)
```py
CHANNEL_ID = ...
GUILD_ID = ...
client = discord.Client(...)
client.channel = client.get_partial_messageable(CHANNEL_ID, guild_id=GUILD_ID)
```
---
Example for `ext.task`
```py
from discord.ext import tasks

...

@tasks.loop(seconds=1800)
async def auto_bump():
    await client.channel.send('/bump')

...

auto_bump.start()
client.run(token)
```
---
It is also advice to make proper commands instead of parsing message content
